azavea/raster-foundry
# Add Explicit Owner Field/Filter

## Overview

This PR adds a new `owner` field to most objects that have a concept of an owner and uses it during the import process. Before this PR, Raster Foundry code implicity identified the `createdBy` user as the "owner"; however, this caused issues when machine users were responsible for creating objects via the API.

This was most apparent when importing imagery. The import would succeed, imagery would be added, but the "owner" would be the airflow user that processed the imagery and a user could not see their own imagery.

### Checklist


### Demo

Imported scene showing up in filter bar

![test-data](https://cloud.githubusercontent.com/assets/898060/25393377/4c81bf66-29a9-11e7-93a6-e17004a3a6e2.png)


## Testing Instructions
 * Switch to branch, run migrations (they take a few minutes)
 * Obtain a test tif of planet imagery (faster if it is clipped, because full resolution still runs into a few issues with footprint polygon simplification)
 * Start airflow server (`docker-compose -f docker-compose.airflow.yml up`) and ensure that the find scenes to import and import geotiff scenes DAGs are the only two that are active
![test-airflow](https://cloud.githubusercontent.com/assets/898060/25393514/a5ceb2fe-29a9-11e7-9c9f-b0333a015765.png)
 * Watch import status, ensure that it completes successfully
 * Go to Add scenes UI for your project, filter based on `Planetscope` and then verify that your imported imagery shows up. Also inspect the owner and ensure that you are the owner, but the created by and modified by should be the airflow user
 * In an API console, check that the following interactions work as expected for `/api/tool-tags/` after obtaining your user ID:

`POST` the following (should succeed):
```
{
  "organizationId": "dfac6307-b5ef-43f7-beda-b9f208bb7726",
  "tag": "User Test"
}
```
The response should include you as the owner

`POST` the following, this time setting you as the owner explicitly (should succeed):
```
{
  "organizationId": "dfac6307-b5ef-43f7-beda-b9f208bb7726",
  "tag": "User Test Two",
  "owner": "<YOUR USER ID>"
}
```

`POST` the following, should fail because you are not in the root organization and cannot set a different owner:
```
{
  "organizationId": "dfac6307-b5ef-43f7-beda-b9f208bb7726",
  "tag": "User Test Three",
  "owner": "rf|airflow-user"
}
```
 * Query the scenes endpoint with `/api/scenes/?owner=<your-user-id>` query parameter which should limit the results with your single scene

Closes #1505, #1506, #1507, #1508, #1509
